pkg/cloudprovider/cloudprovider.go

```diff
@@ -100,6 +104,37 @@ func New(ctx awscontext.Context) *CloudProvider {
 	}
 }
 
+func (c *CloudProvider) IsNodeDrifted(ctx context.Context, provisioner *v1alpha5.Provisioner, node v1.Node) (bool, error) {
+	instanceTypes, err := c.GetInstanceTypes(ctx, provisioner)
```

Comment:
Thanks for pointing this. 
The instanceTypes are coming from cache in the `instanceTypeProvider.Get`, there are two options:
1. I can expose a `instanceTypeProvider.GetSpecificInstanceType` and will return only that, but in that function it has to get all from the cache and then filter by the specific instance type and return only that. 
2. I don't create additional functionality in the instanceTypeProvider but filter that instanceType here in-line and pass it on to `launchTemplateProvider.GetAMIsForProvider` so we get only get only one ami in the array (but the return type would still be an array, if we dont want to refactor the existing code).

But both of these approaches, still return an array of ami with one element, otherwise we will have to create more functions to get just a single ami, according to me that may be an overkill. Happy to hear your thoughts.  :)

Reply:
My point was more that we can pass in the specific InstanceType into the `GetAmisForProvider` call so that we don't have to pass in all the instance types.

I don't think you should add a new interface method to the cloudprovider. Ideally, the behavior would be:
1. `GetInstanceTypes`
2. Find the instanceType that matches the one labeled on the node
3. `GetAmisForProvider` for that specific instance type

Reply:
Done.